acceptance/analyzer_test.go

```diff
@@ -0,0 +1,503 @@
+					h.AssertNotMatch(t, output, "2222 3333 .+ launch-layer.sha")
+					h.AssertNotMatch(t, output, "2222 3333 .+ launch-layer.toml")
+					h.AssertMatch(t, output, "2222 3333 .+ store.toml")
+				})
+			})
+		})
+
+		when("cache is provided", func() {
+			when("cache image case", func() {
+				it.Before(func() {
+					metadata := flattenMetadata(t, filepath.Join("testdata", "analyzer", "cache_metadata.json"), lifecycle.CacheMetadata{})
+
+					cmd := exec.Command(
+						"docker",
+						"build",
+						"-t", cacheImageRepo,
+						"--build-arg", fmt.Sprintf("metadata=%s", metadata),
+						filepath.Join("testdata", "analyzer", "cache-image"),
+					)
+					h.Run(t, cmd)
+				})
+
+				it.After(func() {
+					removeTestImage(t, cacheImageRepo)
+				})
+
+				it("ignores the cache", func() {
+					cmd := exec.Command(
+						"docker", "run", "--rm",
+						"--mount", "type=bind,source=/var/run/docker.sock,target=/var/run/docker.sock",
+						"--env", "CNB_REGISTRY_AUTH={}", // In practice, we never set this variable in the daemon case. Setting to avoid failure to stat docker config directory when initializing cache.
+						analyzeImage,
+						"/bin/bash", "-c",
+						fmt.Sprintf("%s -daemon -cache-image %s some-image; ls -alR /layers", analyzerPath, cacheImageRepo),
+					)
+					output := h.Run(t, cmd)
+
+					h.AssertNotMatch(t, output, "some-layer.sha")
+					h.AssertNotMatch(t, output, "some-layer.toml")
+				})
+			})
+
+			when("cache directory case", func() {
+				var (
+					cacheDir string
+					err      error
+				)
+
+				it.Before(func() {
+					cacheDir, err = ioutil.TempDir("", "analyzer-acceptance")
+					h.RecursiveCopy(t, filepath.Join("testdata", "analyzer", "cache-dir"), cacheDir)
+					// Some OSes (like macOS) use symlinks for the standard temp dir.
+					// Resolve it so it can be properly mounted by the Docker daemon.
+					cacheDir, err = filepath.EvalSymlinks(cacheDir)
+					h.AssertNil(t, err)
+				})
+
+				it.After(func() {
+					os.RemoveAll(cacheDir)
+				})
+
+				it("uses the provided cache", func() {
+					cmd := exec.Command(
+						"docker", "run", "--rm",
+						"--mount", "type=bind,source=/var/run/docker.sock,target=/var/run/docker.sock",
+						"--volume", fmt.Sprintf("%s:/cache", cacheDir),
+						analyzeImage,
+						"/bin/bash", "-c",
+						fmt.Sprintf("%s -daemon -cache-dir /cache some-image; ls -alR /layers", analyzerPath),
+					)
+					output := h.Run(t, cmd)
+
+					h.AssertMatch(t, output, "/layers/some-buildpack-id")
+					h.AssertMatch(t, output, "some-layer.sha")
+					h.AssertMatch(t, output, "some-layer.toml")
+				})
+
+				when("the provided cache directory isn't writeable", func() {
+					it("recursively chowns the directory", func() {
+						cmd := exec.Command(
+							"docker", "run", "--rm",
+							"--mount", "type=bind,source=/var/run/docker.sock,target=/var/run/docker.sock",
+							"--volume", fmt.Sprintf("%s:/cache", cacheDir),
```

Comment:
This issue probably doesn't affect the tests as written b/c the image always running with `USER=root`(and dropping privs) but it might matter (and confuse us) if we change the container user in future test cases.

Reply:
Latest code will copy the cache fixture dir to a docker volume and pass that as `--volume` instead.